src/components/qrcode.tsx

```diff
@@ -0,0 +1,33 @@
+import React, { ReactElement, useEffect, useRef } from "react";
+import qrcode, { QRCodeRenderersOptions } from "qrcode";
+
+interface QrCodeProps {
+  value: string;
+  hasBorder?: boolean;
+}
+
+export const QrCode = ({
+  value,
+  hasBorder,
+  ...rest
+}: QrCodeProps & QRCodeRenderersOptions): ReactElement => {
+  const canvasRef = useRef<HTMLCanvasElement>(null);
+
+  useEffect(() => {
+    if (!canvasRef.current || !value) return;
+
+    qrcode.toCanvas(
+      canvasRef.current,
+      value,
+      { scale: 3, ...rest },
```

Comment:
Is there a strong reason to use `scale` instead of `width`?

Cause we wouldn't know what `3` would translate to in the renderer.

Reply:
Not a strong reason just that having a scale without width makes the QR code render proportionally. (see below) .
The scale defines the width and height of each tiny square in the QR code.(in the case of scale 3, each square will be 3x3px). By having a width just messes up the proportion , each mini square could be 2X3 instead. 

![image](https://user-images.githubusercontent.com/9956949/166611581-122b8000-ca8a-4f8a-93ae-f27305113610.png)

Your point of defining width is a good, hence I am ok to add back widths.

Reply:
Oh then the scale might make more sense then! Let's leave it ☺️

Reply:
okie lets try scale first.